.github/workflows/run-dev-tests.yml

```diff
@@ -104,47 +105,13 @@ jobs:
 
   coverage-report:
     needs: [test]
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: Download all artifacts
-        uses: actions/download-artifact@v4
-        with:
-          path: ci-artifacts
-          pattern: coverage-data-*
-
-      - run: mv ci-artifacts/**/.coverage* ./
-
-      - name: Setup python
-        uses: actions/setup-python@v5
-        with:
-          python-version: '3.x'
-
-      - name: Create coverage report
-        run: |
-          pip3 install coverage[toml]
-          coverage combine
-          coverage html
-
-      - name: Upload comprehensive coverage HTML report
-        uses: actions/upload-artifact@v4
-        with:
-          name: coverage-report
-          path: htmlcov/
-
-      - run: coverage report && coverage xml
-
-      - uses: codecov/codecov-action@v4
-        env:
-          CODECOV_TOKEN: ${{secrets.CODECOV_TOKEN}}
-        with:
-          files: ./coverage.xml
-          fail_ci_if_error: true # optional (default = false)
-          verbose: true # optional (default = false)
-
-      - name: Run codacy-coverage-reporter
-        uses: codacy/codacy-coverage-reporter-action@v1
-        with:
-          project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
-          coverage-reports: ./coverage.xml
+    uses: cpp-linter/.github/.github/workflows/py-coverage.yml@main
+  # Can not get coverage.xml from above reusable workflows.
```

Comment:
I don't really care about uploading coverage reports to codacy; it was just an experiment of  mine. But If we wanted to, then the reusable workflow _could_ upload the coverage.xml as an artifact that we could use after the reusable workflow is finished.